### PR TITLE
Enable eslint rule for missing jsx keys

### LINKED ISSUE #2399: Fail early if keys are missing from an array of JSX elements
Would be nice to have something e.g. a lint rule to prevent keys being missed from elements provided via a loop: https://facebook.github.io/react/docs/lists-and-keys.html#keys

Maybe there's an existing lint available to catch it, or some other way to fail harder earlier?

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #2399

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -83,12 +83,13 @@
     "react/prefer-stateless-function": "off",
     "react/jsx-indent-props": "off",
     "react/jsx-closing-bracket-location": "off",
+    // FIXME: Switch using .jsx for jsx files https://github.com/mozilla/addons-frontend/issues/969.
```
Comment (kumar303) [inline .eslintrc]: It looks like this was copied from the old code and I don't think it's relevant. Can it be removed?

File: src/admin/containers/AddonPage/index.js [hunk 2]
```diff
@@ -33,19 +33,22 @@ class AddonPage extends React.Component {
     ];
     if (addon.homepage) {
       items.push([
+        // eslint-disable-next-line react/jsx-key
```
Comment (kumar303) [inline src/admin/containers/AddonPage/index.js]: Why can't these ones get keys?

File: src/ui/components/Rating/index.js [hunk 3]
```diff
@@ -71,6 +71,7 @@ export class RatingBase extends React.Component {
       }
 
       return (
+        // eslint-disable-next-line react/jsx-key
```
Comment (kumar303) [inline src/ui/components/Rating/index.js]: Ugh. I was afraid of this. There is a `key` defined in the `props` object but it looks like the linter isn't able to recognize it. Could you file an upstream bug?

File: tests/client/amo/containers/TestApp.js [hunk 4]
```diff
@@ -92,6 +92,7 @@ describe('App', () => {
         return <p>The component</p>;
       }
     }
+    // eslint-disable-next-line react/jsx-key
     const root = render({ children: [<MyComponent />] });
```
Comment (kumar303) [inline tests/client/amo/containers/TestApp.js]: You could also just add `key="key"` here.

File: src/admin/containers/AddonPage/index.js (hunk 2 above)
Comment (muffinresearch) [inline src/admin/containers/AddonPage/index.js]: It's actually using data in the object to make the key later on. The eslint rule just sees any jsx in an array as needing a key so this is a false positive.

File: .eslintrc (hunk 1 above)
Comment (muffinresearch) [inline .eslintrc]: 👍

File: src/ui/components/Rating/index.js (hunk 3 above)
Comment (muffinresearch) [inline src/ui/components/Rating/index.js]: One was already filed here: https://github.com/yannickcr/eslint-plugin-react/issues/613

**Pull request merged**